cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -126,6 +135,9 @@ public Executor(KafkaCruiseControlConfig config,
            long removalHistoryRetentionTimeMs) {
     _numExecutionStopped = new AtomicInteger(0);
     _numExecutionStoppedByUser = new AtomicInteger(0);
+    _executionStoppedByUser = new AtomicBoolean(false);
```

Comment:
`_numExecutionStopped` and `_numExecutionStoppedByUser` are not tied to a single execution, and they are used in for gauge sensors in `registerGaugeSensors()`, so maybe not put them as local variable in `ProposalExecutionRunnable`. `_executionStoppedByUser` is a boolean flag and is toggled per execution job to indicate whether the `ProposalExecutionRunnable` is stopped, so I think it could be an instance variable of `Executor`.

Reply:
Sorry I was not clear in original comment. Actually I mean `_executionStoppedByUser` , `_executionException` and `_executionStartMs`, from the code they are only get read when we are sending the notification at the end of execution and they are actually one-time thing(i,e. once execution is done and we send the notification, we do not need these information any more). Therefore I feel it is more appropriate thing to make them [local variable of ProposalExecutionRunnable](https://github.com/linkedin/cruise-control/blob/2b98cdbe4380159b912dcbcc5a4e3082f52c1843/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java#L433)

Reply:
Yeah it makes sense to put `_executionException` and `_executionStartMs` in the Runnable. For `_executionStoppedByUser`, since it's used in tandem with `_stopRequested`, I think we should keep it as a field of `Executor.